### PR TITLE
feat(login-via-google):user can login via google

### PR DESCRIPTION
**Pull request by Nadinefiona**
#### What does this PR do?

This PR implements the functionality for users to log in to the app using their Google accounts. 

#### Description of Task to be completed?

This task involves adding Google login functionality to the app's backend, enhancing user experience and security. It includes documenting the feature with Swagger, developing an algorithm for Google OAuth integration, updating the user model, creating API endpoints, and ensuring at least 80% test coverage. 

#### How should this be manually tested?

1. npm run install
2. npm run dev
3. then swagger for testing 

#### Any background context you want to provide?

This feature enhances the app's user experience by providing a convenient way for users to log in using their Google accounts, streamlining the registration and login process. It also leverages Google's security infrastructure to protect user credentials.

#### What are the relevant pivotal tracker/Trello stories?

-Story ID: 123456789

### CODE REVIEW

File: src/database/migrations/20240325172706-create-user.js [hunk 1]
```diff
@@ -28,6 +28,24 @@ module.exports = {
...
+      },
+      photoUrl: {
+        type: Sequelize.STRING,
+        allowNull: true,
+      },
+      verified: {
+        type: Sequelize.BOOLEAN,
+        allowNull: false,
+        defaultValue: false,
+      },
```
Comment (fab-ryan) [inline src/database/migrations/20240325172706-create-user.js]: please create new migration

**Review by fab-ryan — CHANGES_REQUESTED** [review]
please follow eveyrthing and you are storing content that are not needed like profile what's the meaning of profile

File: src/database/migrations/20240410162214-add-auth-table.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      },
+      googleId: {
+        type: Sequelize.STRING,
+        allowNull: true,
+      },
+      photoUrl: {
+        type: Sequelize.STRING,
+        allowNull: true,
+      },
+      verified: {
```
Comment (fab-ryan) [inline src/database/migrations/20240410162214-add-auth-table.js]: why creating new table we are already have user table

File: src/database/models/auth.ts [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+  declare googleId: string;
+
+  declare photoUrl: string;
+
+  declare email: string;
+
+  declare phone: string;
+
+  declare address: string;
```
Comment (fab-ryan) [inline src/database/models/auth.ts]: this is useless file we have user model

File: src/routers/index.ts [hunk 4]
```diff
@@ -2,9 +2,11 @@ import { Router } from 'express';
 import UserRouter from './userRoute';
 import ProductRouter from './productRoute';
 import rolerouter from './role';
+import authRouter from './authRoute';
 
 const router = Router();
-const routers: Router[] = [UserRouter, ProductRouter];
+const routers: Router[] = [UserRouter, ProductRouter, authRouter];
+
 router.use('/api', routers);
```
Comment (fab-ryan) [inline src/routers/index.ts:10]: please follow the structure of current codes

**Review by fab-ryan — CHANGES_REQUESTED** [review]
Please don't create new file for user why team are not communicating

File: src/app.ts [hunk 5]
```diff
@@ -1,12 +1,24 @@
...
 import * as swaggerDocument from './swagger.json';
 
+// Import and execute Passport setup
+// eslint-disable-next-line import/no-unresolved
+import './config/passport-setup';
+
 dotenv.config();
 const app = express();
+
+app.use(session({ secret: 'scrt', resave: false, saveUninitialized: true }));
```
Comment (fab-ryan) [inline src/app.ts]: please use create team defined in config file

File: src/app.ts [hunk 6]
```diff
@@ -1,12 +1,24 @@
...
+app.use(passport.initialize());
+app.use(passport.session());
+
```
Comment (fab-ryan) [inline src/app.ts]: can you clarify why did you use session??

File: src/routers/authRoute.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import { Router } from 'express';
+import * as authController from '../controllers/authController';
+
+const router = Router();
+
+router.get('/users/google', authController.initiateGoogleLogin);
```
Comment (fab-ryan) [inline src/routers/authRoute.ts]: if you say /google what it mean at least can be google-auth or login-google but you must follow what you have written in docs, precisely that endpoint understoondable

**Review by fab-ryan — CHANGES_REQUESTED** [review]
-  ***first the naming of commit is not following the instruction we can't know if it's feature , chore, or bug***

- work on those comments
- squash those commits

File: src/controllers/authController.ts [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+        createdAt: user.createdAt,
+        updatedAt: user.updatedAt,
+      };
+      try {
+        const token = await userToken(user.id, user.email);
+        // console.log('token', token);
+
+        res.json({ user: userObject, token });
+      } catch (error) {
+        return res.status(500).json({ error: 'Failed to generate token' });
```
Comment (fab-ryan) [inline src/controllers/authController.ts]: please remove userObject we don't need it

**Review by fab-ryan — CHANGES_REQUESTED** [review]
squash those commits to one or two

**Review by Mukunzijames — APPROVED** [review]

**Review by fab-ryan — APPROVED** [review]

**Pull request merged**